contracts/ExchangeRates.sol

```diff
@@ -340,6 +343,33 @@ contract ExchangeRates is Owned, MixinSystemSettings, IExchangeRates {
         return _effectiveValueAndRates(sourceCurrencyKey, sourceAmount, destinationCurrencyKey);
     }
 
+    function effectiveAtomicValueAndRates(
+        bytes32 sourceCurrencyKey,
+        uint sourceAmount,
+        bytes32 destinationCurrencyKey
+    )
+        external
+        view
+        returns (
+            uint value,
+            uint sourceRate,
+            uint destinationRate
+        )
+    {
+        // TODO:
+        //   1. convert into equivalent assets via system settings
+        //   2. call Keep3rV1OracleUSD.assetToAsset()
+        //   3. get internal rates via _effectiveValueAndRates()
+        //   4. get P_CLBUF
+        //   5. compare values from 2. and 4.
+        //   6. obtain rates?
+        //   7. return
+        //
+        //   - do we need to output rates (more painful to calculate when sUSD is not in the pair)?
+        //     if we avoid the circuit breaker with atomic exchanges, in theory we could sanity
+        //     check with trade output rather than rates directly
```

Comment:
Related to the circuit breaker, if we don't need to check it then there's no need to calculate rates here.

Reply:
You mean if we don't have to return source and dest rates? You're correct their purpose is for SIP-65 (the decentralized circuit breaker). 

This SIP proposes only trades into or out of sUSD, so it's manageable to return the tuple for now (and thus keep the circuit breaker), though it would need revisiting if and when synth-to-synth trading was allowed